lib/index.js

```diff
@@ -31,85 +12,110 @@ class PluginPassportOAuth {
-    this.config = Object.assign(defaultConfig, customConfig);
+  async init (config, context) {
+    this.config = Object.assign({}, this.defaultConfig, config);
+    this.context = context;
 
     if (!this.config.strategies || Object.keys(this.config.strategies).length === 0) {
-      /*eslint no-console: 0*/
-      console.warn('[kuzzle-plugin-auth-passport-oauth] No strategies specified.');
-      return Promise.resolve();
+      this.context.log.warn('No strategies specified.');
+      return;
     }
 
-    this.context = context;
-
-    Object.keys(this.config.strategies).forEach(key => {
+    for (const key of Object.keys(this.config.strategies)) {
       this.scope[key] = this.config.strategies[key].scope;
-    });
-
-    return this.context.accessors.storage.bootstrap(storageMapping)
-      .then(() => {
-        this.strategies = {};
-        Object.keys(this.config.strategies).forEach(strategy => {
-          if (!this.config.strategies[strategy].credentials) {
-            throw new this.context.errors.BadRequestError(`Error loading strategy [${strategy}]: no credentials provided`);
-          }
+    }
 
-          try {
-            this.authenticators[strategy] = require('passport-' + (this.config.strategies[strategy].passportStrategy||strategy)).Strategy;
-          } catch (err) {
-            throw new this.context.errors.BadRequestError(`Error loading strategy [${strategy}]: ${err.message}`);
-          }
+    await this.context.accessors.storage.bootstrap(this.storageMappings);
 
-          this.strategies[strategy] = {
-            config: {
-              strategyOptions: this.config.strategies[strategy].credentials,
-              authenticateOptions: {
-                scope: this.config.strategies[strategy].scope
-              },
-              fields: this.config.strategies[strategy].persist
-            },
-            methods: {
-              create: 'create',
-              delete: 'delete',
-              exists: 'exists',
-              getInfo: 'getInfo',
-              update: 'update',
-              validate: 'validate',
-              verify: 'verify',
-              afterRegister: 'afterRegister',
-              getById: 'getById'
-            }
-          };
-
-          // This snippet simply suppresses a warning emitted by Kuzzle during
-          // plugin initialization.
-          // See https://github.com/kuzzleio/kuzzle/pull/1145
-          if (semver.lt(this.context.config.version, '1.4.0')) {
-            this.strategies[strategy].config.constructor = this.authenticators[strategy];
-          } else {
-            this.strategies[strategy].config.authenticator = strategy;
+    await this._initStrategies();
+  }
+
+  async _initStrategies () {
+    const promises = [];
+
+    for (const [name, strategy] of Object.entries(this.config.strategies)) {
+      if (!strategy.credentials) {
+        throw new this.context.errors.BadRequestError(`Error loading strategy [${name}]: no credentials provided`);
```

Comment:
(nitpicking because those lines where there before your modifications) since those errors are thrown at startup, they aren't related to any request, and can be plain Error objects. We tend to use the `assert` module nowadays to perform these verifications.